themes/absimple/absimple.theme.sh

```diff
@@ -0,0 +1,384 @@
+    #    fi
+    if [[ -n $1 ]]; then
+        bg=$(bg_color $1)
+        codes=("${codes[@]}" $bg)
+        debug "Added $bg as background to codes"
+    fi
+    if [[ -n $2 ]]; then
+        fg=$(fg_color $2)
+        codes=("${codes[@]}" $fg)
+        debug "Added $fg as foreground to codes"
+    fi
+
+    debug "Right Codes: "
+    # declare -p codes
+
+    # right always has a separator
+    # if [[ $CURRENT_RBG != NONE && $1 != $CURRENT_RBG ]]; then
+    #     $CURRENT_RBG=
+    # fi
+    declare -a intermediate2=($(fg_color $1) $(bg_color $CURRENT_RBG) )
+    # PRIGHT="$PRIGHT---"
+    debug "pre prompt " $(ansi_r intermediate2[@])
+    PRIGHT="$PRIGHT$(ansi_r intermediate2[@])$RIGHT_SEPARATOR"
+    debug "post prompt " $(ansi_r codes[@])
+    PRIGHT="$PRIGHT$(ansi_r codes[@]) "
+    # else
+    #     debug "no current BG, codes is $codes[@]"
+    #     PRIGHT="$PRIGHT$(ansi codes[@]) "
+    # fi
+    CURRENT_RBG=$1
+    [[ -n $3 ]] && PRIGHT="$PRIGHT$3"
+}
+
+######################################################################
+## Emacs prompt --- for dir tracking
+# stick the following in your .emacs if you use this:
+
+# (setq dirtrack-list '(".*DIR *\\([^ ]*\\) DIR" 1 nil))
+# (defun dirtrack-filter-out-pwd-prompt (string)
+#   "dirtrack-mode doesn't remove the PWD match from the prompt.  This does."
+#   ;; TODO: support dirtrack-mode's multiline regexp.
+#   (if (and (stringp string) (string-match (first dirtrack-list) string))
+#       (replace-match "" t t string 0)
+#     string))
+# (add-hook 'shell-mode-hook
+#           #'(lambda ()
+#               (dirtrack-mode 1)
+#               (add-hook 'comint-preoutput-filter-functions
+#                         'dirtrack-filter-out-pwd-prompt t t)))
+
+prompt_emacsdir() {
+    # no color or other setting... this will be deleted per above
+    PR="DIR \w DIR$PR"
+}
+
+######################################################################
+## Main prompt
+
+build_prompt() {
+    [[ ! -z ${AG_EMACS_DIR+x} ]] && prompt_emacsdir
+    prompt_status
+    [[ -z ${AG_NO_HIST+x} ]] && prompt_histdt
+    #[[ -z ${AG_NO_CONTEXT+x} ]] && prompt_context
+    prompt_virtualenv
+    prompt_dir
+    prompt_git
+    prompt_end
+}
+
+# from orig...
+# export PS1='$(ansi_single $(text_effect reset)) $(build_prompt) '
+# this doesn't work... new model: create a prompt via a PR variable and
+# use that.
+
+set_bash_prompt() {
+    RETVAL=$?
+    PR=""
+    PRIGHT=""
+    CURRENT_BG=NONE
+    PR="$(ansi_single $(text_effect reset))"
+    build_prompt
+
+		PS1=""
+		# date randomly or once per hour
+		if (( $(shuf -i 1-20 -n 1 --random-source=/dev/urandom) == 1 )) ; then #TK || (($PSDATE != $(date +%H))) ; then 
+			PS1+="\$(date +%a) $(date +%Y-%m-%d) "
+		fi
+		PSDATE=$(date +%H)
+		# ... and time
+		PS1+="$(date +%H:%M:%S) "
+
+    # uncomment below to use right prompt
+    #     PS1='\[$(tput sc; printf "%*s" $COLUMNS "$PRIGHT"; tput rc)\]'$PR
+    PS1+=$PR
+}
+
+PROMPT_COMMAND=set_bash_prompt
```

Comment:
This can override the existing settings of `PROMPT_COMMAND`. Please use `safe_append_prompt_command` as the other themes do.

In addition, as we are trying to add `bash-preexec` which hacks `PROMPT_COMMAND`, we might need to add adjustments to [`safe_append_prompt_command`](https://github.com/ohmybash/oh-my-bash/blob/4db7436384e0ddfa62911a101b69dd03a0df53fe/themes/base.theme.sh#L508). Could you investigate the interaction between `safe_append_prompt_command` and `bash-preexec`?

Reply:
today, i'll be missing this and the related query about moving the bash-preexec as a plug-in...

Reply:
This is still not fixed.

Reply:
Fixed.